data_processing.ipynb

```diff
@@ -748,6 +759,421 @@
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>...</th>\n",
+       "      <td>...</td>\n",
+       "      <td>...</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>8</th>\n",
+       "      <td>[0.02746254970056796, 0.022406735048318086, 0....</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>107</th>\n",
+       "      <td>[0.02350585332112601, 0.025473906998611087, 0....</td>\n",
+       "      <td>2</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>490</th>\n",
+       "      <td>[0.02423272928835719, 0.02342685552305181, 0.0...</td>\n",
+       "      <td>2</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>28</th>\n",
+       "      <td>[0.015755475703704988, 0.01818003315805917, 0....</td>\n",
+       "      <td>3</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>[0.02230706281282789, 0.021776700824309162, 0....</td>\n",
+       "      <td>0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "<p>153 rows × 2 columns</p>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                                                    fv label\n",
+       "51   [0.014029436354815213, 0.014280474969277254, 0...     0\n",
+       "370  [0.03026395156144915, 0.030441104676146632, 0....     1\n",
+       "456  [0.018051205585293762, 0.020033789914877715, 0...     2\n",
+       "102  [0.025893012259715704, 0.021900611378626946, 0...     0\n",
+       "36   [0.019407836535601047, 0.0187561280211469, 0.0...     0\n",
+       "..                                                 ...   ...\n",
+       "8    [0.02746254970056796, 0.022406735048318086, 0....     0\n",
+       "107  [0.02350585332112601, 0.025473906998611087, 0....     2\n",
+       "490  [0.02423272928835719, 0.02342685552305181, 0.0...     2\n",
+       "28   [0.015755475703704988, 0.01818003315805917, 0....     3\n",
+       "4    [0.02230706281282789, 0.021776700824309162, 0....     0\n",
+       "\n",
+       "[153 rows x 2 columns]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "3    53\n",
+      "0    37\n",
+      "2    33\n",
+      "1    30\n",
+      "Name: label, dtype: int64\n"
+     ]
+    }
+   ],
+   "source": [
+    "def split_GPAC_data():\n",
+    "    \"\"\"\n",
+    "    Function that takes all of the data, filters out the rows w/ a label of -1, and then splits \n",
+    "    the data into Training, Validation, and Test \n",
+    "    \"\"\"\n",
+    "    \n",
+    "    all_data = pd.DataFrame(data=None, columns=['fv', 'label'])\n",
+    "    for dataset in gpac_image_data:\n",
+    "        \n",
+    "        # Isolate the image data portion of each dataset\n",
+    "        sub_data = gpac_image_data[dataset]['data']\n",
```

Comment:
Is `dataset` an integer? Why is it used as an index into gpac_image_data?

Reply:
gpac_image_data is the dict of all the data, indexed by the path to the data. example : key = data/GPAC_Images_Labelled/data_WY_1_3_export_resample.mat.

I iterate over it so I can perform the split on the entire dataset after its all loaded in.

Reply:
Oh yeah, that's right. Thanks. Maybe you should name it data_file or something, instead of dataset.

Reply:
okioki sounds good